src/pages/resultsView/enrichments/AlterationEnrichmentsContainer.tsx

```diff
@@ -0,0 +1,142 @@
+import * as React from 'react';
+import * as _ from "lodash";
+import { observer } from "mobx-react";
+import { ResultsViewPageStore } from "../ResultsViewPageStore";
+import { observable, computed } from 'mobx';
+import AlterationEnrichmentTable from 'pages/resultsView/enrichments/AlterationEnrichmentsTable';
+import { AlterationEnrichment } from 'shared/api/generated/CBioPortalAPIInternal';
+import styles from "./styles.module.scss";
+import { getAlterationScatterData, getFilteredData, getAlterationRowData } from 'pages/resultsView/enrichments/EnrichmentsUtil';
+import { Checkbox, Button, Form, FormGroup, ControlLabel, FormControl } from 'react-bootstrap';
+import { MolecularProfile } from 'shared/api/generated/CBioPortalAPI';
+import DataSetDropdown from 'pages/resultsView/enrichments/DataSetDropdown';
+import { AlterationEnrichmentRow } from 'shared/model/AlterationEnrichmentRow';
+import MiniScatterChart from 'pages/resultsView/enrichments/MiniScatterChart';
+import MiniBarChart from 'pages/resultsView/enrichments/MiniBarChart';
+import AddCheckedGenes from 'pages/resultsView/enrichments/AddCheckedGenes';
+
+export interface IAlterationEnrichmentContainerProps {
+    data: AlterationEnrichment[];
+    totalAlteredCount: number;
+    totalUnalteredCount: number;
+    headerName: string;
+    store: ResultsViewPageStore;
+}
+
+@observer
+export default class AlterationEnrichmentContainer extends React.Component<IAlterationEnrichmentContainerProps, {}> {
+
+    @observable mutualExclusivityFilter: boolean = true;
+    @observable coOccurenceFilter: boolean = true;
+    @observable significanceFilter: boolean = false;
+    @observable.shallow checkedGenes: string[] = [];
+    @observable clickedGene: string;
+    @observable selectedGenes: string[]|null;
+
+    constructor(props: IAlterationEnrichmentContainerProps) {
+        super(props);
+        this.mutualExclusivityFilterChange = this.mutualExclusivityFilterChange.bind(this);
```

Comment:
you can install react-autobind, which lets you do binding with a decorator on method.   adam and i started using this but it's not merged yet.